sashimi-webapp/test/unit/specs/database/create/tableCreator.spec.js

```diff
@@ -0,0 +1,163 @@
+        isUsed = false;
+        done();
+      }
+    });
+
+    it('should be allowed to add header', (done) => {
+      if (!isUsed) {
+        isUsed = true;
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER');
+        tableCreator.addHeader('b', 'STRING');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.endCreateTable()
+        .catch(err => expect(err).to.not.equal(Error));
+        isUsed = false;
+        done();
+      }
+    });
+
+    it('should be allowed to set foreign key', (done) => {
+      if (!isUsed) {
+        isUsed = true;
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER');
+        tableCreator.addHeader('b', 'STRING');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.endCreateTable()
+        .catch(sqlErr => done(sqlErr))
+        .then(() => {
+          tableCreator.initCreateTable('cba');
+          tableCreator.addHeader('a', 'NUMBER');
+          tableCreator.addHeader('b', 'STRING');
+          tableCreator.addHeader('c', 'DATE');
+          tableCreator.setForeignKey('b', 'abc', 'b');
+          tableCreator.endCreateTable()
+          .catch(err => expect(err).to.not.equal(Error));
+          done();
+          isUsed = false;
+        });
+      }
+    });
+
+    it('should be allowed to set primary key', (done) => {
+      if (!isUsed) {
+        isUsed = true;
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER');
+        tableCreator.addHeader('b', 'STRING');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.setPrimaryKeys('a', 'b', 'c');
+        tableCreator.endCreateTable()
+        .catch(err => expect(err).to.not.equal(Error));
+
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER', 'PRIMARY KEY');
+        tableCreator.addHeader('b', 'STRING');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.endCreateTable()
+        .catch(err => expect(err).to.not.equal(Error));
+        isUsed = false;
+        done();
+      }
+    });
+
+    it('should be allowed to set unique key', (done) => {
+      if (!isUsed) {
+        isUsed = true;
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER');
+        tableCreator.addHeader('b', 'STRING');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.setUnique('a', 'b', 'c');
+        tableCreator.endCreateTable()
+        .catch(err => expect(err).to.not.equal(Error));
+
+        tableCreator.initCreateTable('abc');
+        tableCreator.addHeader('a', 'NUMBER', 'PRIMARY KEY');
+        tableCreator.addHeader('b', 'STRING', 'UNIQUE');
+        tableCreator.addHeader('c', 'DATE');
+        tableCreator.setUnique('a', 'b', 'c');
+        tableCreator.endCreateTable()
+        .catch(err => expect(err).to.not.equal(Error));
+        done();
```

Comment:
noted >.< will improve on the test case. I guess this applies same to key component test as well. Major problem of async in the main component test because alasql couldn't implement delete very well.